Add debugging virt stack (#725)
kubevirt/user-guide: docs/debug_virt_stack/logging.md

# Control libvirt logging for each component

Generally, cluster admins can control the log verbosity of each KubeVirt component in KubeVirt CR. For more details, please, check the [KubeVirt documentation](https://kubevirt.io/user-guide/operations/debug/#log-verbosity).

> Nonetheless, regular users can also adjust the qemu component logging to have a more fine control over it. The annotation `kubevirt.io/libvirt-log-filters` enables you to modify each component's log level.

Reviewer: ```suggestion
Nonetheless, regular users can also adjust the qemu component logging to have finer control over it. The annotation `kubevirt.io/libvirt-log-filters` enables you to modify each component's log level.
```

Example:
```yaml
apiVersion: kubevirt.io/v1
kind: VirtualMachineInstance
metadata:
  annotations:
    kubevirt.io/libvirt-log-filters: "2:qemu.qemu_monitor 3:*"
  labels:
    special: vmi-debug-tools
  name: vmi-debug-tools
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: containerdisk
      - disk:
          bus: virtio
        name: cloudinitdisk
      rng: {}
    resources:
      requests:
        memory: 1024M
  volumes:
  - containerDisk:
      image: registry:5000/kubevirt/fedora-with-test-tooling-container-disk:devel
    name: containerdisk
  - cloudInitNoCloud:
      userData: |-
        #cloud-config
        password: fedora
        chpasswd: { expire: False }
    name: cloudinitdisk
```

Then, it is possible to obtain the logs from the virt-launcher output:

```bash

> $ kubectl  get po

Reviewer: Nice shortcut, that'll save me 2 keystrokes many times a day :)
However, the full version might look better in docs?
```suggestion
$ kubectl get pods
```

{"component":"virt-launcher","level":"info","msg":"QEMU_MONITOR_RECV_EVENT: mon=0x7faa8801f5d0 event={\"timestamp\": {\"seconds\": 1698324644, \"microseconds\": 165626}, \"event\": \"VSERPORT_CHANGE\", \"data\": {\"open\": true, \"id\": \"channel0\"}}","pos":"qemuMonitorJSONIOProcessLine:205","subcomponent":"libvirt","thread":"80","timestamp":"2023-10-26T12:50:44.165000Z"}
[..]
{"component":"virt-launcher","level":"info","msg":"QEMU_MONITOR_RECV_EVENT: mon=0x7faa8801f5d0 event={\"timestamp\": {\"seconds\": 1698324646, \"microseconds\": 707666}, \"event\": \"RTC_CHANGE\", \"data\": {\"offset\": 0, \"qom-path\": \"/machine/unattached/device[8]\"}}","pos":"qemuMonitorJSONIOProcessLine:205","subcomponent":"libvirt","thread":"80","timestamp":"2023-10-26T12:50:46.708000Z"}
[..]
```

The annotation enables the filter from the container creation. However, in

> certain case you might desire to change the logging level dynamically once the container and libvirt have already been started. In this case, [`virt-admin`](https://libvirt.org/manpages/virt-admin.html) come to rescue.

Reviewer: ```suggestion
certain cases you might desire to change the logging level dynamically once the container and libvirt have already been started. In this case, [`virt-admin`](https://libvirt.org/manpages/virt-admin.html) comes to the rescue.
```

Example:
```console
$ kubectl get pods
NAME                                READY   STATUS    RESTARTS   AGE
virt-launcher-vmi-ephemeral-nqcld   3/3     Running   0          26m
$ kubectl  exec -ti virt-launcher-vmi-ephemeral-nqcld -- virt-admin  -c virtqemud:///system  daemon-log-filters  "1:libvirt 1:qemu 1:conf 1:security 3:event 3:json 3:file 3:object 1:util"

> $ kubectl  exec -ti virt-launcher-vmi-ephemeral-nqcld -- virt-admin  -c virtqemud:///system  daemon-log-filters

Author: Yes, you are right!

> $ kubectl  exec -ti virt-launcher-vmi-ephemeral-nqcld -- virt-admin  -c virtqemud:///system  daemon-log-filters

Author: @vasiliy-ul thanks for pointing this our. It is weird while it is working with both `-c virtqemud:///system` and `-c virtqemud:///session` in a rootless pod. But session is for sure correct

 Logging filters: 1:*libvirt* 1:*qemu* 1:*conf* 1:*security* 3:*event* 3:*json* 3:*file* 3:*object* 1:*util* 
```

> Otherwise, if you prefer to redirect the output to a file and fetch it later, you can rely on `kubectl cp` to retrieve the file. In this case, we are saving the file in the `/var/run/libvirt` directory because the compute container has the permissions to write on that path.

Reviewer: ```suggestion
Otherwise, if you prefer to redirect the output to a file and fetch it later, you can rely on `kubectl cp` to retrieve the file. In this case, we are saving the file in the `/var/run/libvirt` directory because the compute container has write permissions there.
```
